specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2023-04-01-preview/ContentPackages.json

```diff
@@ -0,0 +1,486 @@
+              "$ref": "../../../common/2.0/types.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.OperationalInsights/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/contentPackages/{packageId}": {
+      "get": {
+        "x-ms-examples": {
+          "Get installed packages by id.": {
+            "$ref": "./examples/contentPackages/GetPackageById.json"
+          }
+        },
+        "tags": [
+          "ContentPackages"
+        ],
+        "description": "Gets an installed packages by its id.",
+        "operationId": "ContentPackages_Get",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../common/2.0/types.json#/parameters/WorkspaceName"
+          },
+          {
+            "$ref": "#/parameters/PackageIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/packageModel"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/2.0/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      },
+      "put": {
+        "x-ms-examples": {
+          "Install a package to the workspace.": {
+            "$ref": "./examples/contentPackages/InstallPackage.json"
+          }
+        },
+        "tags": [
+          "ContentPackages"
+        ],
+        "description": "Install a package to the workspace.",
+        "operationId": "ContentPackage_Install",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../common/2.0/types.json#/parameters/WorkspaceName"
+          },
+          {
+            "$ref": "#/parameters/PackageIdParameter"
+          },
+          {
+            "$ref": "#/parameters/PackageInstallationProperties"
```

Comment:
>PackageInstallationProperties [](http://example.com/codeflow?start=34&length=29)

Since packageId is part of "id" property of packageModel anyway, why is packageId property needed explicitly needed in packageModel? That would eliminate the need of separate model for request and response.

Also, ARM requires RPs to have same model for request and response in PUT APIs.

Reply:
Please check [my previous comment](https://github.com/Azure/azure-rest-api-specs/pull/23151#discussion_r1152352715)

Reply:
@visingla-ms 
Please check my latest comment. Thank you!

Reply:
See my other comment on this.